server/datastore/mysql/migrations/tables/20230315104937_EnsureUniformCollation.go

```diff
@@ -0,0 +1,73 @@
+package tables
+
+import (
+	"database/sql"
+	"fmt"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20230315104937, Down_20230315104937)
+}
+
+// changeCollation changes the default collation set of the database and all
+// table to the provided collation
+//
+// This is based on the changeCharacterSet function that's included in this
+// module and part of the 20170306075207_UseUTF8MB migration.
+func changeCollation(tx *sql.Tx, charset string, collation string) error {
+	_, err := tx.Exec(fmt.Sprintf("ALTER DATABASE DEFAULT CHARACTER SET %s COLLATE %s", charset, collation))
+	if err != nil {
+		return fmt.Errorf("alter database: %w", err)
+	}
+
+	rows, err := tx.Query(`
+          SELECT table_name
+          FROM information_schema.TABLES AS T, information_schema.COLLATION_CHARACTER_SET_APPLICABILITY AS C
+          WHERE C.collation_name = T.table_collation
+          AND T.table_schema = (SELECT database())
+          AND (C.CHARACTER_SET_NAME != ? OR C.COLLATION_NAME != ?)`, charset, collation)
+	if err != nil {
+		return fmt.Errorf("selecting tables: %w", err)
+	}
+	defer rows.Close()
+
+	var names []string
+	for rows.Next() {
+		var name string
+		if err := rows.Scan(&name); err != nil {
+			return fmt.Errorf("scanning ID: %w", err)
+		}
+		names = append(names, name)
+	}
+
+	if err := rows.Err(); err != nil {
+		return fmt.Errorf("scanning rows: %w", err)
+	}
+
+	// disable foreign checks before changing the collations, otherwise the
+	// migration might fail. These are re-enabled afterwards.
+	if _, err := tx.Exec("SET FOREIGN_KEY_CHECKS = 0"); err != nil {
+		return fmt.Errorf("disabling foreign key checks: %w", err)
+	}
+	for _, name := range names {
+		_, err = tx.Exec(fmt.Sprintf("ALTER TABLE %s CONVERT TO CHARACTER SET %s COLLATE %s", name, charset, collation))
```

Comment:
Might want to use the backticks just to be safe (around the `%s`).

Reply:
good catch, thanks. I also added backticks  to the first `tx.Exec`